### PR TITLE
Adding support for IDLE command

### PR DESCRIPTION
**Pull request by davidgfnet**
I want to get some feedback on implementing IDLE command for IMAP.
This is just a simple proof of concept, I am missing:
 - Proper timeout handling: do not close connection on timeouts for IDLE command
 - Use IDLE/DONE sequences: Schedule a DONE+IDLE sequence every N seconds to keepalive TCP & IMAP connection (and other possible sources of disconnect such as NAT).
 - Proper reporting to the user: The idea is that the user will perform a custom IDLE command to avoid continuous fetches to the inbox.
 - Add/check support for aborting the transaction: If the user implements a progress callback we should be able to gracefully terminate the IDLE operation.

I'm waiting for your feedback :D 
Thanks!

### CODE REVIEW

File: lib/imap.c [hunk 1]
```diff
@@ -1807,6 +1862,30 @@ static CURLcode imap_sendf(struct connectdata *conn, const char *fmt, ...)
 
 /***********************************************************************
  *
+ * imap_sendr()
+ *
+ * Sends the formated string as araw string
```
Comment (dmelani) [inline lib/imap.c]: s/araw/a raw/

File: lib/imap.c [hunk 2]
```diff
@@ -813,6 +819,26 @@ static CURLcode imap_perform_search(struct connectdata *conn)
...
+ * imap_perform_idle()
+ *
+ * Sends an IDLE command.
+ */
+static CURLcode imap_perform_idle(struct connectdata *conn)
+{
+  CURLcode result = CURLE_OK;
+  struct IMAP *imap = conn->data->req.protop;
+
+  /* Send the SEARCH command */
```
Comment (jay) [inline lib/imap.c]: s/SEARCH/IDLE

**Comment by jay** [comment]
Hi, I like the idea of adding IDLE but unfortunately as you can see there doesn't seem to be any interest for this. My understanding of IDLE is it allows for a push notification from the server. What is your use case here, go IDLE and wait for an e-mail? I think if you are interested in pursuing this some code that shows how one might use this in a program would be helpful.

**Comment by davidgfnet** [comment]
Hey!
Yep, the idea is that you send an IDLE command and curl waits until something happens on the server side. I can create some examples. Could you detail exactly what do you want? :D Thanks for the interest!
We might need to add other commands (within an FSM maybe) to keep connection alive too. The idea is to use this along with async I/O, otherwise it just blocks forever and you can't do things in the background (you cannot even cancel the command!).

**Comment by jay** [comment]
Basically some client code that we can use to understand what you intend and whether it will be a good fit within libcurl, so you don't waste your time on something that may not be a good fit or that nobody wants to support. Interest here seems to be limited. There are a lot of good feature ideas but then once they are added they need to be maintained.

For example libcurl sends an IDLE and then waits for the push notification. I suppose in its current state one would call easy or multi and wait for the header callback to be called with the push notification, and then what? I also think it would be good to explain what you mean by async, because at the moment one can call easy_perform and it's blocking but if one calls multi_perform that would effectively block as well for as long as there are reads and writes for it to do. In other words multi_perform could theoretically stay out indefinitely iirc, but of course the callbacks will be called when appropriate. So maybe you have one thread using libcurl waiting for idle reply and then it does what?

**Comment by bagder** [comment]
Seems abandoned. Feel free to re-open on new activity/responses.

**Pull request closed without merging**